lib/react-popout.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+let CONTAINER_ID = 'popout-content-container';
+
+/**
+ * @class PopoutWindow
+ */
+export default class PopoutWindow extends React.Component {
+
+  /**
+   * @type {{title: *, url: *, onClosing: *, options: *, window: *, containerId: *}}
+   */
+  static propTypes = {
+    title: React.PropTypes.string.isRequired,
+    url: React.PropTypes.string,
+    onClosing: React.PropTypes.func,
+    options: React.PropTypes.object,
+    window: React.PropTypes.object,
+    containerId: React.PropTypes.string
+  }
+
+	/**
+   * @constructs PoppoutWindow
+   * @param props
+   */
+  constructor(props){
+    super(props);
+
+    this.defaultOptions = {
+      toolbar: 'no',
+      location: 'no',
+      directories: 'no',
+      status: 'no',
+      menubar: 'no',
+      scrollbars: 'yes',
+      resizable: 'yes',
+      width: 500,
+      height: 400,
+      top: (o, w) => ((w.innerHeight - o.height) / 2) + w.screenY,
+      left: (o, w) => ((w.innerWidth - o.width) / 2) + w.screenX
+    };
+
+    this.state = {
+      openedWindow: null
+    };
+  }
+
+  /**
+   * Override default id if we get given one
+   * @param props
+   */
+  componentWillReceiveProps(props){
+    props.containerId && (CONTAINER_ID = props.containerId);
```

Comment:
this is an instance level change that has static level repercussions. If you have 2 popouts on the page the last one will win.